lib/WWW/Spotify.pm

```diff
@@ -410,7 +441,38 @@ sub send_get_request {
     if ( $self->grab_response_header() == 1 ) {
         $self->_set_response_headers($mech);
     }
-    return $self->format_results( $mech->content );
+
+    $self->response_status( $mech->status() );
+    $self->response_content_type( $mech->content_type() );
+
+    if ( ref( $self->request_custom_handler() ) eq 'CODE' ) {
+        $self->request_custom_handler_result(
+            $self->request_custom_handler()->($mech) );
+    }
+
+    # the original code did not provide adequate bulit in validation
+    # of the response for an API call.
+    # Adding a new method (check_response) with a deault of 0 to avoid
+    # breaking/changing
+    # existing code using older versions of this module.
+    # verify the status and content_type of the response
+    if (   $self->response_content_type() =~ /application\/(json|xml)/i
+        && $self->response_status() != '200' ) {
```

Comment:
For status codes validation, you can use `HTTP::Status` as well, which is already in the dependencies.

```perl
use HTTP::Status qw( HTTP_OK );
```

and then later

...
if ( ... && $self->response_status() != HTTP_OK ) {
```